lib/plugins/acts_as_customizable/lib/acts_as_customizable.rb

```diff
@@ -116,6 +119,35 @@ def reset_custom_values!
           custom_values.each { |cv| cv.destroy unless custom_field_values.include?(cv) }
         end
 
+        def validate_custom_values
+          custom_values.reject { |cv| cv.marked_for_destruction? }.each do |custom_value|
+            unless custom_value.valid?
+              custom_value.errors.each do |_, message|
+                errors.add(custom_value.custom_field.accessor_name.to_sym, message)
+              end
+            end
+          end
+        end
+
+        def add_custom_field_accessors
+          available_custom_fields.each do |custom_field|
+            getter_name = custom_field.accessor_name
+            setter_name = "#{getter_name}="
+
+            define_singleton_method getter_name do
+              custom_value = custom_value_for(custom_field)
+              custom_value ? custom_value.typed_value : nil
+            end
+
+            define_singleton_method setter_name do |value|
+              # N.B. we do no strict type checking here, it would be possible to assign a user
+              # to an integer custom field...
+              value = value.id if value.respond_to?(:id)
+              self.custom_field_values = { custom_field.id => value }
+            end
+          end
+        end
```

Comment:
I have thought about this again and I fear that we might add a performance impediment here. Imagine you get a list of WorkPackages. For every WorkPackage the list of CustomFields would have to be fetched from the DB. That cannot be fast.

So I'd now opt for the method_missing variant you favoured initially already. We can still add the method once it has been found missing so the code is not lost.